filebeat/module/elasticsearch/audit/ingest/pipeline-plaintext.json

```diff
@@ -51,6 +51,28 @@
                 "field": "elasticsearch.audit.sub_action",
                 "ignore_missing": true
             }
+        },
+        {
+            "date": {
+                "field": "elasticsearch.audit.@timestamp",
+                "target_field": "@timestamp",
+                "formats": [
+                    "yyyy-MM-dd'T'HH:mm:ss,SSS"
+                ],
+                "ignore_failure": true
```

Comment:
Now that this processor and the next one are both acting on the same `field`, perhaps we could add a `"if": "ctx.event.timezone == null"` property to this processor for a bit of extra clarity?

Reply:
I think it's better to not add `"if": "ctx.event.timezone == null"`, beacause:
1. other modules don't have `"if": "ctx.event.timezone == null"` added, we'd better keep this convention
2. in most cases, `event.timezone` was not set. without `"if": "ctx.event.timezone == null"` would make the second processor looks like a `optional` path instead of a `alternative` path

Reply:
I think @ycombinator is right, we could add this `if`, as only one of the processors is needed.

> * other modules don't have `"if": "ctx.event.timezone == null"` added, we'd better keep this convention
> 

In previous versions both processors were meant to be executed, one to parse the date, and the next one to apply the timezone. We (well, you @pragkent :slightly_smiling_face: ) found this approach is not correct in many cases, so now we are duplicating the date processor, first option only parses the date, and second option parses the date with a timezone if available.

> * in most cases, `event.timezone` was not set. without `"if": "ctx.event.timezone == null"` would make the second processor looks like a `optional` path instead of a `alternative` path

Actually we are setting `event.timezone` by default since ~7.2 in all filebeat events, so in most cases both date processors are executed even if only the last one is needed.

Said that, as we had it quite tested by both @pragkent and me, and it solves an existing issue, I'd go on with merging this change as is, and have a follow up PR to review the conditions in the pipelines we have recently changed to fix this same issue. This way we keep a common convention for this.